### ISSUE TITLE
False-positive E1137 (unsupported-assignment-operation) with np.empty_like

### ISSUE THREAD

**Issue by EmilyBourne** [pylint-dev/pylint#9385]
An almost identical error has already been reported (#2767).

The example code from that issue:
```python
"""X"""
import numpy as np

X = np.empty_like([1])
X[:] = 2
```
works as expected, but the almost identical code:
```python
"""X"""
from numpy import empty_like

X = empty_like([1])
X[:] = 2
```
raises the error:
```
************* Module bug
bug.py:5:0: E1137: 'X' does not support item assignment (unsupported-assignment-operation)
```

## pylint --version output
```
pylint 3.0.3
astroid 3.0.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

_Originally posted by @EmilyBourne in https://github.com/pylint-dev/pylint/issues/2767#issuecomment-1884376916_
